### PR TITLE
Allow users to set the googmodule bazelOpt in their tsconfig

### LINKED ISSUE #230: Add support for generating goog modules
Currently the [`tsc_wrapped/compiler_host.ts`](https://github.com/bazelbuild/rules_typescript/blob/master/internal/tsc_wrapped/compiler_host.ts) reads the [`googmodule` property](https://github.com/bazelbuild/rules_typescript/blob/master/internal/tsc_wrapped/compiler_host.ts#L113) from the tsconfig, but it isn't ever set in [tsconfig.bzl](https://github.com/bazelbuild/rules_typescript/blob/master/internal/common/tsconfig.bzl#L122).

In my testing, setting it to `true` causes the [`ngc_wrapped/index.ts`](https://github.com/angular/bazel-builds/blob/master/src/ngc-wrapped/index.ts) compiler to output goog modules, because it uses the `tsc_wrapped/compiler_host.ts` and [emits with tsickle](https://github.com/angular/bazel-builds/blob/master/src/ngc-wrapped/index.ts#L260), but I haven't been able to get [`tsc_wrapped.ts`](https://github.com/bazelbuild/rules_typescript/blob/master/internal/tsc_wrapped/tsc_wrapped.ts) to emit goog modules directly because it appears to use [some other emit](https://github.com/bazelbuild/rules_typescript/blob/master/internal/tsc_wrapped/tsc_wrapped.ts#L145).

### PR DESCRIPTION
**Pull request by alexeagle**
Fixes #230

### CODE REVIEW

**Comment by JaredNeil** [comment]
This should probably also set `bazelOpts.tsickle` to `true`, or also let the `tsickle` option to be set from the `tsconfig.json`.

File: internal/tsc_wrapped/tsc_wrapped.ts [hunk 1]
```diff
@@ -139,15 +140,32 @@ function runOneBuild(
     return false;
   }
 
+  const emitter = bazelOpts.googmodule || bazelOpts.tsickle ? 'tsickle' : 'ts';
```
Comment (mprobst) [inline internal/tsc_wrapped/tsc_wrapped.ts]: wouldn't this be easier as a plain boolean `emitWithTsickle`?

File: internal/tsc_wrapped/tsc_wrapped.ts [hunk 2]
```diff
@@ -139,15 +140,32 @@ function runOneBuild(
...
+  const emitResults: ts.EmitResult[] = [];
```
Comment (mprobst) [inline internal/tsc_wrapped/tsc_wrapped.ts]: maybe make this a `ts.EmitResult[]|tsickle.EmitResult[]`?

Comment (alexeagle) [reply internal/tsc_wrapped/tsc_wrapped.ts]: Doesn't help much, still need to cast it below to narrow out the ts.EmitResult which is missing the `modulesManifest`

**Pull request closed without merging**